airbyte-workers/src/main/java/io/airbyte/workers/general/DefaultNormalizationWorker.java

```diff
@@ -57,10 +64,14 @@ public NormalizationSummary run(final NormalizationInput input, final Path jobRo
 
       if (!normalizationRunner.normalize(jobId, attempt, normalizationRoot, input.getDestinationConfiguration(), input.getCatalog(),
           input.getResourceRequirements())) {
-        throw new WorkerException("Normalization Failed.");
+        normalizationRunner.getTraceMessages()
+            .forEach(traceMessage -> traceFailureReasons.add(FailureHelper.normalizationFailure(traceMessage, Long.valueOf(jobId), attempt)));
+        failed = true;
       }
     } catch (final Exception e) {
-      throw new WorkerException("Normalization Failed.", e);
+      normalizationRunner.getTraceMessages()
+          .forEach(traceMessage -> traceFailureReasons.add(FailureHelper.normalizationFailure(traceMessage, Long.valueOf(jobId), attempt)));
+      failed = true;
```

Comment:
👍 there can be more than one trace message
Not that we have any yet, but there can be trace messages that are *not* failureReasons... do you need to filter this array?

Reply:
good call, added filtering for ERROR traces.